extensions/wikia/Insights/models/InsightsFlagsModel.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/**
+ * Data model specific to a subpage with a list of pages marked with flags
+ */
+class InsightsFlagsModel extends InsightsPageModel {
+	const INSIGHT_TYPE = 'flags';
+
+	public
+		$flagTypeId,
+		$loopNotificationConfig = [
+			'displayFixItMessage' => false,
+		];
+
+	public function getInsightType() {
+		return self::INSIGHT_TYPE;
+	}
+
+	public function getPaginationUrlParams() {
+		if ( $this->flagTypeId ) {
+			return [ 'flagTypeId' => $this->flagTypeId ];
+		}
+		return [];
+	}
+
+	/**
+	 * A key of a message that wraps the number of pages referring to each item of the list.
+	 *
+	 * @return string
+	 */
+	public function wlhLinkMessage() {
+		return 'insights-used-on';
+	}
+
+	public function arePageViewsRequired() {
+		return false;
+	}
+
+	/**
+	 * Get a type of a subpage only, we want a user to be directed to view.
+	 * @return array
+	 */
+	public function getUrlParams() {
+		return $this->getInsightParam();
+	}
+
+	/**
+	 * Get list of articles related to the given QueryPage category
+	 *
+	 * @return array
+	 */
+	public function getContent( $params ) {
+		$this->preparePaginationParams( $params );
+		$this->flagTypeId = $params['flagTypeId'];
+		return $this->fetchArticlesData();
+	}
+
+	/**
+	 * Prepare data of articles - title, last revision, link etc.
+	 * @return array
+	 */
+	public function fetchArticlesData() {
+		$articlesData = [];
+		$flaggedPages = $this->getPagesByFlagType();
+
+		if ( count( $flaggedPages ) > 0 ) {
+			$articlesData = $this->prepareData( $flaggedPages );
+
+			if ( $this->arePageViewsRequired() ) {
+				$articlesIds = array_keys( $articlesData );
+				$pageViewsData = $this->getPageViewsData( $articlesIds );
+				$articlesData = $this->assignPageViewsData( $articlesData, $pageViewsData );
+			}
+		}
+
+		return $articlesData;
+	}
+
+	/**
+	 * @param array $pagesIds Array of pages Ids
+	 * @return array
+	 * e.g. result
+	 * [
+	 * 	{page_id} => [
+	 * 		'link' => {link_to_page},
+	 * 		'metadata' => ['lastRevision'=>{array_with_revision_data} ] @see \InsightsPageModel::prepareRevisionData
+	 * 	]
+	 * ]
+	 * @throws MWException
+	 */
+	public function prepareData( $pagesIds ) {
+		$data = [];
+
+		foreach ( $pagesIds as $pageId ) {
+			$article = [];
+
+			$title = Title::newFromID( $pageId );
```

Comment:
Do we clear Flag entries in the table when a page is deleted? If not, a deleted page would cause the `getTitleLink` below to fatal because this title would return null if the page ID doesn't exist in the page table, i.e. would cause:
` PHP Catchable fatal error:  Argument 1 passed to InsightsHelper::getTitleLink() must be an instance of Title, null given`